net2cog/netcdf_convert.py

```diff
@@ -7,24 +7,22 @@
 Functions related to converting a NetCDF file to other formats.
 """
 
-import logging
 import os
 import pathlib
-import subprocess
-import tempfile
-from os.path import join as pjoin, basename, dirname, exists, splitext
-from subprocess import check_call
+from logging import Logger
+from os.path import join as path_join, basename
+from tempfile import TemporaryDirectory
 from typing import List
 
 import rasterio
 import rioxarray  # noqa
 import xarray as xr
+from harmony_service_lib.util import generate_output_filename
```

Comment:
Is it possible to pass this in somehow? `netcdf_convert` shouldn't have any Harmony dependencies otherwise we need to make this module dependent on harmony instead of including it as an extra. Maybe an option would be a temporary name here and then `netcdf_convert_harmony.py` does the rename to a standard?

Reply:
I'm going to resolve this thread and track the overall issue of migrating the `harmony-service-lib` functionality out of `netcdf_convert.py` in [this thread](https://github.com/podaac/net2cog/pull/38#discussion_r1939874329).